### PR TITLE
Fix release doc refs

### PR DESCRIPTION
**Pull request by kohr-h**
Bad references in docs. I temporarily changed the `.travis.yml` to ensure that the docs are built for this PR, will remove it again when it succeeds.

### CODE REVIEW

**Comment by kohr-h** [comment]
Previous builds failed because `sphinx` would output such a massive amount of warnings that the log would become too big (> 4 Mb) and Travis would bail out. This current change breaks the docs in bad ways, so that's not an option either. 

Piping in `.travis.yml` is not allowed, so we cannot filter the warnings on the console without wrapping it into some kind of custom script.
An alternative would be to simply disable all warnings in sphinx (or at least all that can be silenced) and check manually once in a while. Also kinda ugly.

Currently there are ~32000 warnings, we need to do something about it. For reference:

- A bunch of actual errors
- Several `WARNING: document isn't included in any toctree`
- Some `WARNING: Citation [XYZ] is not referenced`¨
- Many `WARNING: more than one target found for 'any' cross-reference`
- TONS of `<autosummary>:: WARNING: toctree contains reference to nonexisting document`
  Basically one such warning for each class and each attribute as it seems

File: doc/source/conf.py [hunk 1]
```diff
@@ -91,11 +90,12 @@ def setup(app):
 autosummary_generate = glob.glob("./*.rst")
 
 # Default role for ambiguous ':any:' targets
-default_role = 'py:class'
+default_role = ':py:class:'
 
 # Stops WARNING: toctree contains reference to nonexisting document
 numpydoc_show_class_members = True
 numpydoc_show_inherited_class_members = True
+numpydoc_class_members_toctree = False  # this should avoid a bunch of warnings
```
Comment (kohr-h) [inline doc/source/conf.py]: but also breaks other things

File: doc/source/conf.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def setup(app):
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
-exclude_patterns = ['*.py', '*.pyc']
+exclude_patterns = ['*.py', '*.pyc', 'doc/source/generated']
```
Comment (kohr-h) [inline doc/source/conf.py]: this was just a desperate attempt, breaks stuff of course

File: doc/source/conf.py [hunk 3]
```diff
@@ -91,11 +90,12 @@ def setup(app):
 autosummary_generate = glob.glob("./*.rst")
 
 # Default role for ambiguous ':any:' targets
-default_role = 'py:class'
+default_role = ':py:class:'
 
 # Stops WARNING: toctree contains reference to nonexisting document
```
Comment (kohr-h) [inline doc/source/conf.py]: No it doesn't

**Comment by kohr-h** [comment]
For now the sphinx build redirects stderr to /dev/null so the log doesn't explode in size.
